### PR TITLE
ipatests: add tests for cached_auth_timeout in sssd.conf

### PR DESCRIPTION
**Pull request by wladich**
The tests check that auth cache
    * is disabled by default
    * is working when enabled
    * expires after specified time
    * is inherited by trusted domain
    
    Related to: https://bugzilla.redhat.com/1685581

### CODE REVIEW

**Comment by flo-renaud** [comment]
@wladich 
I am adding the postponed label to this PR to temporarily prevent it from triggering PRCI.
It seems that PRCI is unable to handle the RunADTests class and crashes because of this.

File: ipatests/prci_definitions/temp_commit.yaml [hunk 1]
```diff
@@ -53,14 +57,14 @@ jobs:
         timeout: 1800
         topology: *build
 
-  fedora-30/temp_commit:
+  fedora-30/test_sssd:
     requires: [fedora-30/build]
     priority: 50
     job:
-      class: RunPytest
+      class: RunADtests
```
Comment (flo-renaud) [inline ipatests/prci_definitions/temp_commit.yaml]: should be:
```
     class: RunADTests
```
with an uppercase T after AD. Otherwise PRCI crashes

Comment (wladich) [reply ipatests/prci_definitions/temp_commit.yaml]: Thanks for comment

File: ipatests/pytest_ipa/integration/tasks.py [hunk 2]
```diff
@@ -1872,3 +1873,57 @@ def run_command_as_user(host, user, command, *args, **kwargs):
...
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.restore()
+
+
+class RemoteIniFile(configparser.RawConfigParser):
+    """Wrap RawConfigParser for modifying .ini files on remote hosts.
```
Comment (flo-renaud) [inline ipatests/pytest_ipa/integration/tasks.py]: This class does not really inherit the RawConfigParser class. For instance read() and read_file() methods probably won't work unless you override them.
I am wondering whether object composition wouldn't be a better fit here.

**Comment by wladich** [comment]
Changes done in new commit:
* updated nightly_master_testing.yaml with new test suite
* replaced tasks.RemoteIniFile with more simple remote_ini_file contexyt manager
* updated docstring for tasks.FileBackup
* added --preserve=context in FileBackup constructor and restore()

File: ipatests/pytest_ipa/integration/tasks.py (hunk 2 above)
Comment (wladich) [reply ipatests/pytest_ipa/integration/tasks.py]: I replaced this class with a simple context manager.

File: ipatests/pytest_ipa/integration/tasks.py [hunk 3]
```diff
@@ -1872,3 +1873,57 @@ def run_command_as_user(host, user, command, *args, **kwargs):
...
+    def __init__(self, host, filename):
+        """Create file backup."""
+        self._host = host
+        self._filename = filename
+        self._backup = create_temp_file(host)
+        host.run_command(['cp', filename, self._backup])
+
+    def restore(self):
+        """Restore file. Can be called multiple times."""
+        self._host.run_command(['cp', self._backup, self._filename])
```
[the remark this answers is not in the document]
Comment (wladich) [reply ipatests/pytest_ipa/integration/tasks.py]: Good catch, fixed.

**Comment by flo-renaud** [comment]
Hi @wladich 
Thanks for the PR, LGTM. You can remove the temp commit and I will ACK once PRCI completes.

**Comment by wladich** [comment]
I have noticed that latest changes to tasks.py were added to the wrong commit. 
I rearranged changes by commits, overall diff did not change.

**Comment by flo-renaud** [comment]
Tests are green, ACK.

**Comment by wladich** [comment]
master:

* 7dde3a422067c82b67eab197a0da3d6f149c1b3c ipatests: add new utilities for file management
* 4ea9aead5cab96ed9454facdb6d22bdfe514a09b ipatests: refactoring: use library function to check if selinux is enabled
* 4ab2842b76ffc9cffd931bf652d211ca34b4d1d8 ipatests: add tests for cached_auth_timeout in sssd.conf

**Pull request closed without merging**